lib/md051.js

```diff
@@ -6,7 +6,8 @@ const { addError, escapeForRegExp, filterTokens, forEachInlineChild,
   forEachHeading, htmlElementRe } = require("../helpers");
 
 // Regular expression for identifying HTML anchor names
-const identifierRe = /(?:id|name)\s*=\s*['"]?([^'"\s>]+)/iu;
+const identifierRe = /id\s*=\s*['"]?([^'"\s>]+)/iu;
+const nameRe = /name\s*=\s*['"]?([^'"\s>]+)/iu;
```

Comment:
Should we put \w at the beginning to block "data-id" from matching?

Reply:
I noticed that too. Done and added a unit test.

(I think you meant `\s`. Note that `\b` doesn't work because "-" is not a word character.)

Reply:
Right, I meant word boundary, but good call on '-'.